tls-options/client-tls-peerauth.c

```diff
@@ -0,0 +1,337 @@
+
+static WC_INLINE int myVerify(int preverify, WOLFSSL_X509_STORE_CTX* store)
+{
+    char buffer[WOLFSSL_MAX_ERROR_SZ];
+    WOLFSSL_X509* peer;
+    char issuerBuffer[CERT_BUFFER_SZ];
+    char subjectBuffer[CERT_BUFFER_SZ];
+
+    fprintf(stderr, "In verification callback, error = %d, %s\n", store->error,
+                                wolfSSL_ERR_error_string(store->error, buffer));
+    peer = store->current_cert;
+    if (peer) {
+        wolfSSL_X509_NAME_oneline(
+            wolfSSL_X509_get_issuer_name(peer), issuerBuffer, CERT_BUFFER_SZ);
+        wolfSSL_X509_NAME_oneline(
+            wolfSSL_X509_get_subject_name(peer), subjectBuffer, CERT_BUFFER_SZ);
+        printf("\tPeer's cert info:\n issuer : %s\n subject: %s\n",
+                                                issuerBuffer, subjectBuffer);
+    }
+    else
+        fprintf(stderr, "\tPeer has no cert!\n");
+
+    printf("\tSubject's domain name at %d is %s\n",
+            store->error_depth, store->domain);
+
+    switch (myVerifyAction) {
+        case VERIFY_OVERRIDE_ERROR:
+            return 1;
+
+        case VERIFY_FORCE_FAIL:
+            return 0;
+
+        case VERIFY_DEFAULT:
+        case VERIFY_USE_PREVERIFY:
+            return preverify;
+        
+        case VERIFY_OVERRIDE_DATE_ERR:
+            if (store->error == ASN_BEFORE_DATE_E
+                    || store->error == ASN_AFTER_DATE_E)
+                return 1;
+            return 0;
+    }
+    fprintf(stderr, "Invalid verify action.\n");
+    exit(1);
+}
+
+int switchPeerAuthMode(char *mode)
+{
+    if (XSTRCMP("NONE", mode) == 0) {
+        return SSL_VERIFY_NONE;
+    }
+    else if (XSTRCMP("PEER", mode) == 0) {
+        return SSL_VERIFY_PEER;
+    }
+    else if (XSTRCMP("FAIL_IN_NO_PEER_CERT", mode) == 0) {
+        return SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
+    }
+    else if (XSTRCMP("FAIL_EXCEPT_PSK", mode) == 0) {
+        return SSL_VERIFY_FAIL_EXCEPT_PSK;
+    }
+    fprintf(stderr, "Invalid peer auth mode.\n");
+    exit(1);
+}
+
+int switchMyVerifyAction(char *mode)
+{
+    if (XSTRCMP("OVERRIDE_ERROR", mode) == 0) {
+        return VERIFY_OVERRIDE_ERROR;
+    }
+    else if (XSTRCMP("FORCE_FAIL", mode) == 0) {
+        return VERIFY_FORCE_FAIL;
+    }
+    else if (XSTRCMP("USE_PREVERIFY", mode) == 0) {
+        return VERIFY_USE_PREVERIFY;
+    }
+    else if (XSTRCMP("OVERRIDE_DATE_ERR", mode) == 0) {
+        return VERIFY_OVERRIDE_DATE_ERR;
+    }
+    fprintf(stderr, "Invalid verify action.\n");
+    exit(1);
+}
+
+int main(int argc, char** argv)
+{
+    int                sockfd = SOCKET_INVALID;
+    struct sockaddr_in servAddr;
+    char               buff[256];
+    size_t             len;
+    int                ret;
+    int                diff = 0;
+    int                peerAuthMode = WOLFSSL_VERIFY_DEFAULT;
+
+    /* declare wolfSSL objects */
+    WOLFSSL_CTX* ctx = NULL;
+    WOLFSSL*     ssl = NULL;
+
+    // /* Check options */
```

Comment:
Please remove `//`

Reply:
I'm sorry for being late.
There was a problem with the git tree, and it was resolved just before the consecutive holidays, so it has not been reflected yet.
I will push the appropriate tree today or tomorrow.
Thank you for your confirmation.